_overviews/FAQ/index.md

```diff
@@ -110,28 +110,56 @@ One would hope so, but doing it that way was tried and it proved
 
-### Difference between methods and functions
+### What's the difference between methods and functions?
+
+For example, how does a method such as:
+
+    def square(x: Int): Int = x * x
+
+differ from a function value such as:
+
+    val square: Int => Int = x => x * x
 
 [complete answer on Stack Overflow](https://stackoverflow.com/a/2530007/4111404)
 
 [summary with practical differences](https://tpolecat.github.io/2014/06/09/methods-functions.html)
 
-### Difference between classes and types
+### What's the difference between types and classes?
+
+Types are primarily a compile-time concept. At compile time,
+every expression is assigned a type by the compiler.
+
+Classes are primarily a runtime concept and are platform-dependent.
+At runtime on the JVM, every value is either a primitive value
+or an instance of exactly one class.
+
+Some type information exists only at compile time, for multiple
+reasons, most notoriously [type
+erasure](https://en.wikipedia.org/wiki/Type_erasure).
+
+For an in-depth treatment of types vs. classes, see the blog post
+["There are more types than classes"](https://typelevel.org/blog/2017/02/13/more-types-than-classes.html).
+
+### How can a method in a superclass return a value of the “current” type?
 
-[blog post: There are more types than classes](https://typelevel.org/blog/2017/02/13/more-types-than-classes.html)
+Possible solutions include F-bounded polymorphism (familiar to Java programmers),
+type members, and the typeclass pattern.
 
-### How to return "current" type
+[discussion of alternatives on Stack Overflow](https://stackoverflow.com/questions/59813323/advantages-of-f-bounded-polymorphism-over-typeclass-for-return-current-type-prob)
 
-[discussion about alternatives and their trade-offs on Stack Overflow](https://stackoverflow.com/questions/59813323/advantages-of-f-bounded-polymorphism-over-typeclass-for-return-current-type-prob)
+[blog post](http://tpolecat.github.io/2015/04/29/f-bounds.html) arguing against F-bounds and in favor of typeclasses.
 
-### What does <:< means?
+### What does `<:<` mean?
 
-[blog post: Generalized type constraints](https://blog.bruchez.name/2015/11/generalized-type-constraints-in-scala.html)
+It's a "type constraint", and it comes from the standard library,
+not from the language itself.  See [this blog
+post](https://blog.bruchez.name/2015/11/generalized-type-constraints-in-scala.html).
 
-### Best way to model optional arguments
+### I dislike requiring callers to wrap optional arguments in `Some(...)`; is there a better way?
 
-[answer on Stack Overflow](https://stackoverflow.com/a/65256691/4111404)
+Not really. See [this answer on Stack Overflow](https://stackoverflow.com/a/65256691/4111404).
 
-### Why prfer implicit val over implicit object
+### Why is `implicit val` usually recommended over `implicit object`?
 
-[answer on Stack Overflow](https://stackoverflow.com/a/65258340/4111404)
+The latter has a singleton type, which is too specific.  See [answer on Stack
```

Comment:
Maybe a link to something that explains what is a singleton type?
Or maybe that link should be in the SO answer itself?

Reply:
Sadly the language tour doesn't have a "Singleton" types section. I agree there's room for improvement here, but I'm not going to tackle it right now.